getExclusions.m

```diff
@@ -11,49 +10,52 @@
                 tcs.Move.(types{1}));
 nUnits = tmp(:, 1);                             % how many units per area                    
 
-% preallocate
-for ia = 1:nAreas
-   OK.isVisual{ia} = zeros(nTypes, nUnits(ia));   
-   OK.anyVisual{ia} = zeros(1, nUnits(ia));   
-   OK.DSI{ia} = zeros(nTypes, nUnits(ia));
-end
-
-
 % Must respond more to move than blank throughout experiment
 %   in at least one of the 12 directions
 
 % parameters 
 pThresh = 0.05;
-nBins = 4;
-
+trialStep = 10; %test 10 trials at a time for visual responsiveness (move > blank)
 
 for ia = 1:nAreas   
 for it = 1:nTypes
-    overThresh = zeros(nUnits(ia), nDirs);
+    p = nan(nUnits(ia), nBins, nDirs);
 for id = 1:nDirs
     tmpMV = tcs.Move.(types{it}){ia,id};
     tmpBK = tcs.Blank.(types{it}){ia,id};
     
     nTrials = size(tmpBK, 2);
-    trialStep = nTrials/nBins;
+    nBins = ceil(nTrials/trialStep);
     binStarts = 1:trialStep:nTrials;
+    
+    %preallocate
+    if it == 1 && id == 1
+        OK.byTime{ia} = zeros(nTypes, nUnits(ia), nTrials);
+        OK.isVisual{ia} = zeros(nTypes, nUnits(ia));   
+        OK.anyVisual{ia} = zeros(1, nUnits(ia));   
+        OK.DSI{ia} = zeros(nTypes, nUnits(ia));
+    end
+    
     for iu = 1:nUnits(ia)
-        p = zeros(1, nBins);
         for ib = 1:nBins
             trialRange = binStarts(ib):binStarts(ib)+trialStep-1;
-            p(ib) = kruskalwallis([tmpBK(iu,trialRange)' ...
-                                   tmpMV(iu,trialRange)'], [], 'off');
+            trialRange = trialRange(trialRange <= length(tmpBK)); %last bin may have fewer trials
+            p(iu, ib, id) = ranksum(tmpBK(iu,trialRange), ...
+                                    tmpMV(iu,trialRange), 'tail', 'left');
         end
         % if all bins are significantly modulated
         if prod(p < pThresh) ~= 0 
```

Comment:
if all(p<pThresh) will save you 2 nanoseconds. 
More importantly - do we care about multiple comparisons? I guess this can be quite permissive.

Reply:
Line 43 tells me that p is 3D. So prod(p<pThresh) is 2D. 
So I'm confused what the output if line 47 actually gives you.

Reply:
Both of these are corrected now.